src/Storages/StatisticsDescription.cpp

```diff
@@ -115,8 +114,11 @@ void ColumnStatisticsDescription::merge(const ColumnStatisticsDescription & othe
 
 void ColumnStatisticsDescription::assign(const ColumnStatisticsDescription & other)
 {
+    /// If the statistics is empty, it's possible that we have not assign a column_name.
+    if (empty() && column_name == "")
+        column_name = other.column_name;
     if (other.column_name != column_name)
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot assign statistics from column {} to {}", column_name, other.column_name);
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot assign statistics from column {} to {}", other.column_name, column_name);
```

Comment:
I think in `ColumnStatisticsDescription::fromAST` we always fill in `name` and `ColumnStatisticsDescription::fromAST` is called in `AlterCommand::apply`, so I am not sure how we end up with a `ColumnStatisticsDescription` without a filled in `name`.

Reply:
Maybe just remove the `column_name` from `ColumnStatisticsDescription` and pass it in as a const reference to `assign` and `merge`? Actually merge is already doing it, so I think it should be easier. It is only used for logging as far as I see.

Reply:
yes it's possible. I removed `column_name`